src/Microsoft.Azure.SignalR.Common/Auth/AzureActiveDirectoryOptions.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Security.Cryptography.X509Certificates;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Azure.SignalR
+{
+    public class AzureActiveDirectoryOptions
+    {
+        public readonly X509Certificate2 ClientCert;
+
+        public string ClientId { get; }
+
+        public string ClientSecret { get; }
+
+        public string TenantId { get; }
+
+        public bool Enabled { get; } = false;
+
+        public string Instance { get; set; } = "https://login.microsoftonline.com/{0}";
+
+        public static readonly AzureActiveDirectoryOptions Disabled = new AzureActiveDirectoryOptions();
+
+        private static readonly Regex GuidRegex = new Regex("^[0-9A-Za-z]{8}-[0-9A-Za-z]{4}-[0-9A-Za-z]{4}-[0-9A-Za-z]{4}-[0-9A-Za-z]{12}$");
+
+        private AzureActiveDirectoryOptions()
+        {
+        }
+
+        private AzureActiveDirectoryOptions(string tenantId)
+        {
+            if (!GuidRegex.IsMatch(tenantId))
+            {
+                throw new FormatException($"{tenantId} is not a valid tenandId, should be [xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx]");
+            }
+            TenantId = tenantId;
+        }
+
+        public Uri BuildAuthority()
+        {
+            var s = string.Format(Instance, TenantId);
+            try
+            {
+                return new Uri(s);
+            }
+            catch (Exception)
+            {
+                throw new FormatException($"{Instance} is invalid, please refer to [link] for available Instances.");
+            }
+        }
+
+        public AzureActiveDirectoryOptions(string clientId, string clientSecret, string tenantId) : this(tenantId)
+        {
+            ClientId = clientId;
+            ClientSecret = clientSecret;
+            Enabled = true;
+        }
+
+        public AzureActiveDirectoryOptions(string clientId, X509Certificate2 clientCert, string tenantId) : this(tenantId)
```

Comment:
looks like `this(tenantId)` can be `this(clientId, tenantId)` and `Enabled = true;` can be moved into it?